src/mlx/warnings.py

```diff
@@ -13,45 +14,99 @@
 
 
 class WarningsPlugin:
-    def __init__(self):
-        self.sphinx_counter = 0
-        self.doxygen_counter = 0
-        self.junit_counter = 0
 
-    def check_sphinx_warnings(self, line):
+    def __init__(self, sphinx = False, doxygen = False, junit = False):
```

Comment:
Yeah, I was thinking about this indeed. I will make it more activate checker, since checkers will be defined internally. We can then ofc make a custom regex...